src/internal/hacks.c

```diff
@@ -43,6 +43,25 @@ char* igraph_i_strdup(const char *s) {
     return result;
 }
 
+/**
+ * Drop-in replacement for strndup.
+ * Used only in compilers that do not have strndup or _strndup
+ */
+char *igraph_i_strndup(const char *s1, size_t n) {
+    size_t i;
+    /* We need to check if the string is shorter than n characters.
```

Comment:
Tip: I think `memchr` might be useful here.

Reply:
But also: don't spend more time than necessary. Let's make things work before we make them pretty. A TODO note about memchr() might be good for later.